core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java

```diff
@@ -283,30 +283,21 @@ private int readFromHdfsBuffer() throws IOException {
     return BufferUtils.byteToInt(mBuffer[mBufferPosition++]);
   }
 
-  /**
-   * This method is not supported in {@link HdfsFileInputStream}.
-   *
-   * @param position N/A
-   * @param buffer N/A
-   * @throws IOException always
-   */
   @Override
   public void readFully(long position, byte[] buffer) throws IOException {
-    throw new IOException(ExceptionMessage.NOT_SUPPORTED.getMessage());
+    readFully(position, buffer, 0, buffer.length);
   }
 
-  /**
-   * This method is not supported in {@link HdfsFileInputStream}.
-   *
-   * @param position N/A
-   * @param buffer N/A
-   * @param offset N/A
-   * @param length N/A
-   * @throws IOException always
-   */
   @Override
   public void readFully(long position, byte[] buffer, int offset, int length) throws IOException {
-    throw new IOException(ExceptionMessage.NOT_SUPPORTED.getMessage());
+    int nread = 0;
+    while (nread < length) {
+      int ret = read(position + nread, buffer, offset + nread, length - nread);
+      if (ret == -1) {
```

Comment:
Is this still allowable if you read the EOF before you finish reading the 'length' amount of data?